website/docs/help-and-support/troubleshooting-guide/widget-errors.md

```diff
@@ -1,38 +1,84 @@
 
-### Cause
+#### Cause
 
 The problem can be caused when you are trying to bind:
 
 * A large array of multiple JSON objects
 * A huge JSON object which has a lot of fields
 * The whole query data rather than a selected row or triggered row in a table
 
-### Solution
+#### Solution
 
 To determine if the problem is caused due to:
 
-* **A large array or a huge JSON object** - You can relook at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
-* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either selected row / triggered row to bind.
+* **A large array or a huge JSON object** - You can re-look at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
+* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either the selected row / triggered row to bind.
 
 Once you have figured out the new structure for the data, head to the [source data](../../reference/widgets/json-form#source-data) field to make changes.
 
-:::info
-If you still can't see a JSON form and need help debugging an error, please get in touch with us on our [Discord Server](https://discord.com/invite/rBTTVJp) or email us at support@appsmith.com.
-:::
\ No newline at end of file
+## Query/API/JS object execution errors
+You may see below errors when executing an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) using a widget property.
+
+### Sync fields can't execute async framework actions
+
+You may see this error when binding an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS Object](../../core-concepts/writing-code/javascript-editor-beta/) execution to a widget property that expects data, and can't be used to trigger an action. 
+
+#### Error message
+You would see one of the below error messages based on API/Query or a JS Object function execution.
+
+> Found a reference to {{action}} during evaluation. Sync fields cannot execute async framework actions. Please remove any direct/indirect references to {{action}} and try again.
+
+ or 
+
+> Found a Promise() during evaluation. Sync fields cannot execute asynchronous code.
+
+
+**To-Do to add screenshot**
+
+#### Cause
+**Action** refers to the execution of an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/). 
```

Comment:
API, Query and JS Objects are cross-referenced once above. So we can skip the cross-referencing again here